common/core/gpudevice.cpp

```diff
@@ -201,6 +201,8 @@ void GpuDevice::DisplayManager::HotPlugEventHandler() {
           "UpdateDisplayState.");
       UpdateDisplayState();
     }
+    /* Adding sleep to avoid busy looping before a retry */
+    usleep(1000);
```

Comment:
Lets separate sleep to a separate PR, and we can check the perf impact, ensure it doesnt affect hotplug event in any way and make a call. Rest looks good to be merged.